lib/queue.js

```diff
@@ -726,65 +725,27 @@ Queue.prototype.processJob = function(job){
 
     stopTimer();
 
-    // TODO: Should moveToFailed ensure the lock atomically in one of its Lua scripts?
     // See https://github.com/OptimalBits/bull/pull/415#issuecomment-269744735
-    job.takeLock(true /* renew */, false /* ensureActive */).then( function(/*lock*/) {
-      return job.moveToFailed(err).finally(function(){
-        _this.processing--;
-      }).then(function(){
-        return _this.distEmit('failed', job, error);
-      });
-    }, function(err){
-      _this.emit('error', err, 'failed to re-obtain lock before moving to failed, bailing');
+    //  job.takeLock(true /* renew */, false /* ensureActive */).then( function(/*lock*/) {
+    return job.moveToFailed(err).then(function(){
```

Comment:
I do not think this could happen in practice since the lock extender timer executes twice as often as the lock expiration time. However, if I move it to finally it also passes the unit tests, so I can move it there. Ultimately I will also make moveToFailed atomic, to further minimize any possible hazards.